### PR TITLE
refactor hooks/users.ts into individual files within hooks/users/*

### PR DESCRIPTION
**Pull request by 0xdevalias**
Refactored `hooks/users.ts` into individual files within `hooks/users/*`

### CODE REVIEW

File: src/hooks/users.ts [hunk 1]
```diff
@@ -1,150 +0,0 @@
-import { useMemo } from "react";
```
Comment (0xdevalias) [inline src/hooks/users.ts:1]: [note] This file was moved to `src/hooks/users/index.ts` and then split into individual files under `src/hooks/users/*`

File: src/hooks/users/useConnectWorldUsers.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { useFirestoreConnect } from "hooks/useFirestoreConnect";
+import { useSovereignVenueId } from "hooks/useSovereignVenueId";
+import { useVenueId } from "hooks/useVenueId";
+
+export const useConnectWorldUsers = () => {
```
Comment (0xdevalias) [inline src/hooks/users/useConnectWorldUsers.ts:5]: [note] This hook was split out of `src/hooks/users/index.ts` into it's own file here.

File: src/hooks/users/useRecentLocationUsers.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * @description this hook's filtering world users based on their @lastSeenIn location
+ *
+ * @param locationName is a key for `lastSeenIn` firestore field in user's object
+ *
+ * @example useRecentLocationUsers(venue.name)
+ * @example useRecentLocationUsers(`${venue.name}/${roomTitle}`)
+ */
+export const useRecentLocationUsers = (locationName?: string) => {
```
Comment (0xdevalias) [inline src/hooks/users/useRecentLocationUsers.ts:17]: [note] This hook was split out of `src/hooks/users/index.ts` into it's own file here.

File: src/hooks/users/useRecentVenueUsers.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { useConnectCurrentVenueNG } from "hooks/useConnectCurrentVenueNG";
+import { useVenueId } from "hooks/useVenueId";
+
+import { useRecentLocationUsers } from "./useRecentLocationUsers";
+
+export const useRecentVenueUsers = () => {
```
Comment (0xdevalias) [inline src/hooks/users/useRecentVenueUsers.ts:6]: [note] This hook was split out of `src/hooks/users/index.ts` into it's own file here.

File: src/hooks/users/useRecentWorldUsers.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import { User } from "types/User";
+
+import { WithId } from "utils/id";
+import { normalizeTimestampToMilliseconds } from "utils/time";
+
+import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
+
+import { useWorldUsers } from "./useWorldUsers";
+
+export const useRecentWorldUsers = (): {
```
Comment (0xdevalias) [inline src/hooks/users/useRecentWorldUsers.ts:12]: [note] This hook was split out of `src/hooks/users/index.ts` into it's own file here.

File: src/hooks/users/useWorldUsers.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+import { WithId } from "utils/id";
+import { worldUsersSelector } from "utils/selectors";
+
+import { isLoaded } from "hooks/useFirestoreConnect";
+import { useSelector } from "hooks/useSelector";
+
+import { useConnectWorldUsers } from "./useConnectWorldUsers";
+
+export const useWorldUsers = (): {
```
Comment (0xdevalias) [inline src/hooks/users/useWorldUsers.ts:13]: [note] This hook was split out of `src/hooks/users/index.ts` into it's own file here.

File: src/hooks/users/useWorldUsersById.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+import { isLoaded } from "hooks/useFirestoreConnect";
+import { useSelector } from "hooks/useSelector";
+
+import { useConnectWorldUsers } from "./useConnectWorldUsers";
+
+/**
+ * @debt typing, Record implies that a User will exist for literally any given string, which is untrue
+ * @deprecated use useWorldUsersByIdWorkaround until we refactor this to fix the typing issue across the codebase
+ */
+export const useWorldUsersById = () => {
```
Comment (0xdevalias) [inline src/hooks/users/useWorldUsersById.ts:16]: [note] This hook was split out of `src/hooks/users/index.ts` into it's own file here.

File: src/hooks/users/useWorldUsersById.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+      worldUsersById: worldUsersById ?? {},
+      isWorldUsersLoaded: isLoaded(worldUsersById),
+    }),
+    [worldUsersById]
+  );
+};
+
+// @debt typing, this uses Partial<Record<K,T>> to work around the bug where Record implies that a User will exist for literally any given string, which is untrue
+// @debt refactor all usages of Record<string, User> to Partial<Record<string, User>> (then we can remove this workaround)
+export const useWorldUsersByIdWorkaround = () => {
```
Comment (0xdevalias) [inline src/hooks/users/useWorldUsersById.ts:32]: [note] This hook was split out of `src/hooks/users/index.ts` into it's own file here.

**Review by 0xdevalias — COMMENTED** [review]
Notes

**Comment by goran-peoski-work** [comment]
LGTM

**Review by sunny-viktoryia — APPROVED** [review]
Thanks for the refactoring! Approving. 🙃

**Pull request merged**